src/seedu/addressbook/AddressBook.java

```diff
@@ -198,8 +198,35 @@
      * ====================================================================
      */
     public static void main(String[] args) {
-        showWelcomeMessage();
-        processProgramArgs(args);
+        String[] s1 = new String[5]; 
+        s1[0] = DIVIDER; 
+        s1[1] = DIVIDER; 
+        s1[2] = VERSION; 
+        s1[3] = MESSAGE_WELCOME; 
+        s1[4] = DIVIDER; 
+        for (String m : s1) {
+            System.out.println(LINE_PREFIX + m);
+        }
+        if (args.length >= 2) {
+            System.out.println(LINE_PREFIX + MESSAGE_INVALID_PROGRAM_ARGS);
```

Comment:
Note that the constants are also performing some abstraction. for e.g., If you replaced LINE_PREFIX by the actual prefix everywhere. SLAP is not limited to just methods.